components/core/src/clp_s/SchemaReader.cpp

```diff
@@ -294,6 +302,236 @@ int32_t SchemaReader::get_first_column_in_span(std::span<int32_t> schema) {
     return -1;
 }
 
+void SchemaReader::find_intersection_and_fix_brackets(
+        int32_t cur_root,
+        int32_t next_root,
```

Comment:
I think I'd prefer to keep the cur_root/next_root names. They carry some meaning because they're supposed to indicate the root of the subtree for which we've pushed all of the required keys/brackets to m_json_serializer before the field we are about to push. I'm open to other names that carry meaning though.

Reply:
Got it. Personally I prefer using`...id` for `int32_t` node IDs as we extensively use `...column_id`, `...child_id` elsewhere. But it's fine to leave them as they are since it seems that we are following `...node` for `SchemaNode` and `...root` for `int32_t`.